conformance_tests/sysman/test_sysman_init/src/test_init_sysman.cpp

```diff
@@ -24,4 +244,391 @@ TEST(SysmanInitTests,
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  validate_sysman_api(static_cast<zes_device_handle_t>(ze_devices[0]),
+                      ZE_RESULT_ERROR_UNINITIALIZED);
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanInitializedViaZesInitAndCoreInitializedWithSysmanEnabledWhenSysmanApiAreCalledWithCoreToSysmanMappedHandleThenSuccessIsReturned) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  ze_device_properties_t properties = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+  ASSERT_EQ(ZE_RESULT_SUCCESS,
+            zeDeviceGetProperties(ze_devices[0], &properties));
+
+  zes_uuid_t uuid = {};
+  memcpy(uuid.id, properties.uuid.id, ZE_MAX_DEVICE_UUID_SIZE);
+  std::vector<zes_driver_handle_t> zes_drivers = get_zes_drivers();
+
+  zes_device_handle_t sysman_device_handle{};
+  ze_bool_t on_subdevice = false;
+  uint32_t subdevice_id = 0;
+  EXPECT_EQ(ZE_RESULT_SUCCESS, zesDriverGetDeviceByUuidExp(
+                                   zes_drivers[0], uuid, &sysman_device_handle,
+                                   &on_subdevice, &subdevice_id));
+  ASSERT_NE(sysman_device_handle, nullptr);
+
+  validate_sysman_api(sysman_device_handle, ZE_RESULT_SUCCESS);
+}
+
+TEST(SysmanInitTests, GivenValidDeviceWhenCallingInitThenSuccessIsReturned) {
+  char sysman_env_enable[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sysman_env_enable);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  ze_device_properties_t properties = {};
+  ze_device_ip_version_ext_t ip_version_ext{};
+  properties.stype = ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES;
+  properties.pNext = &ip_version_ext;
+  ip_version_ext.stype = ZE_STRUCTURE_TYPE_DEVICE_IP_VERSION_EXT;
+  ip_version_ext.pNext = nullptr;
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS,
+            zeDeviceGetProperties(ze_devices[0], &properties));
+
+  bool new_platform = false;
+  if (properties.type == ZE_DEVICE_TYPE_GPU && properties.vendorId == 0x8086) {
+    ze_device_ip_version_ext_t *ip_version =
+        static_cast<ze_device_ip_version_ext_t *>(properties.pNext);
+    if (ip_version->ipVersion >= 0x05004000) {
+      new_platform = true;
+    }
+  }
+
+  if (new_platform) {
+    ASSERT_EQ(ZE_RESULT_ERROR_UNINITIALIZED, zesInit(0));
```

Comment:
This should be checking only after checking legacy sysman is enabled. you can call any Enum APIs and make sure of it